src/NServiceBus.Core/Pipeline/PipelineComponent.cs

```diff
@@ -16,7 +17,14 @@ public static PipelineComponent Initialize(PipelineSettings settings, HostingCom
 
             foreach (var registeredBehavior in modifications.Replacements)
             {
-                hostingConfiguration.Services.ConfigureComponent(registeredBehavior.BehaviorType, DependencyLifecycle.InstancePerCall);
+                hostingConfiguration.Services.AddTransient(registeredBehavior.BehaviorType);
+
+                var interfaces = registeredBehavior.BehaviorType.GetInterfaces();
```

Comment:
We should extract all this to a helper since we have to do this in lots of other places